### PR TITLE
Adding swagger docs for existing myworkbooks API's

### PR DESCRIPTION
**Pull request by gardnerjr**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by gardnerjr** [comment]
shouldn't be "Breaking", as this is all existing apis that somehow just didn't have docs here (for years?).  adding them for S360 compliance.  all of the "myworkbooks" stuff is essentially copies of the "workbooks" things that did exist.

**Comment by MyronFanQiu** [comment]
@gardnerjr Please fix CI errors first

**Comment by gardnerjr** [comment]
> @gardnerjr Please fix CI errors first

@myronfanqui do you know what the deal is with `R4007 - DefaultErrorResponseSchema` ? the error object in this schema does have those fields (and more?), and i presume that the other 31 previous errors might be the same response for the /workbooks and other paths that use our error schema object?

is there a way to tag these as having to match the existing behavior as this has been shipped for years?

for `PathResourceProviderNamePascalCase` does it really want `myworkbooks` to be `MyWorkbooks` ? even though everything is supposed to be treated as [case insensitive](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#case-insensitivity-for-requests) anyway?

**Comment by MyronFanQiu** [comment]
@gardnerjr According to http://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html#_Toc372793091, I think it doesn't meet the following requirement. "The error response MUST be a single JSON object. This object MUST have a single name/value pair named error. The value must be a JSON object.". 

Regarding to another rule, the answer is yes. Even though ARM is case insensitive. The following down-streaming products rely on swagger spec and we need to make them share same pattern.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "parameters": {
+    "api-version": "2015-05-01",
+    "subscriptionId": "6b643656-33eb-422f-aee8-3ac145d124af",
+    "resourceGroupName": "my-resource-group",
+    "resourceName": "deadb33f-5e0d-4064-8ebb-1a4ed0313eb2"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "c0deea5e-3344-40f2-96f8-6f8e1c3b5722",
```
Comment (ravbhatnagar) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json:11]: incorrect Ids, it should be the fully qualified ARM resource Id

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        "name": "deadb33f-8bee-4d3b-a059-9be8dac93960",
+        "type": "",
```
Comment (ravbhatnagar) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json:13]: please fix.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        "location": "westus",
+        "properties": {
+          "name": "My New Workbook",
```
Comment (ravbhatnagar) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json:16]: can you call this display Name. top level property cant be repeated inside the properties envelope.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/myworkbooks": {
```
Comment (ravbhatnagar) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:37]: camel case resource type name

**Review by ravbhatnagar — COMMENTED** [review]
few comments, but these may be existing APIs. please fix whatever you can.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json (hunk 3 above)
Comment (gardnerjr) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json:16]: can't rename /change things, these are all existing apis that just weren't properly documented in swagger.  the whole /myworkbooks type is effectively a copy of the /workbooks type (they were even deployed at the same time, not sure why the swagger stuff wasn't done at the same time)

Comment (gardnerjr) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json:16]: actually now that i look at this, it seems like this is out of date and it IS displayName in the real data.  i'm not sure why the workbooks stuff i copied has it wrong too. :/

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json (hunk 2 above)
Comment (gardnerjr) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json:13]: which part?  just that type is empty?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json (hunk 3 above)
Comment (gardnerjr) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/MyWorkbookGet.json:16]: actually this one IS displayname in the actual place, not sure why the workbooks swagger doesn't have that, so i can fix it here

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+        },
+        "serializedData": {
+          "type": "string",
+          "description": "Configuration of this particular private workbook. Configuration data is a string containing valid JSON"
+        },
+        "version": {
+          "type": "string",
+          "description": "This instance's version of the data model. This can change as new features are added that can be marked private workbook."
+        },
+        "kind": {
```
Comment (MyronFanQiu) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:412]: Why do we have two `kind`s?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json [hunk 6]
```diff
@@ -0,0 +1,554 @@
...
+        "timeModified": {
+          "type": "string",
+          "description": "Date and time in UTC of the last modification that was made to this private workbook definition.",
+          "readOnly": true
+        },
+        "category": {
+          "type": "string",
+          "description": "Workbook category, as defined by the user at creation time."
+        },
+        "tags": {
```
Comment (MyronFanQiu) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:435]: It's not same with the `tag` in other resource. It's just a list of string.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json [hunk 7]
```diff
@@ -0,0 +1,554 @@
...
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "A list of 0 or more tags that are associated with this private workbook definition"
+        },
+        "userId": {
+          "type": "string",
+          "description": "Unique user id of the specific user that owns this private workbook."
```
Comment (MyronFanQiu) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:444]: Is it arbitrary string?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json [hunk 8]
```diff
@@ -0,0 +1,554 @@
...
+        },
+        "sourceId": {
+          "type": "string",
+          "description": "Optional resourceId for a source resource."
```
Comment (MyronFanQiu) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:448]: Which kinds of resource are supported?

File: specification/applicationinsights/resource-manager/readme.md [hunk 9]
```diff
@@ -97,6 +97,12 @@ directive:
     where: $.definitions.WorkbookResource.properties.name
     from: workbooks_API.json
     reason: 'This property is already a part of our API, cannot remove it'
+  - suppress: DefaultErrorResponseSchema
+    from: myworkbooks_API.json
+    reason: 'this error format is already part of the previous api, cannot change it'
+  - suppress: DeleteOperationResponses
+    from: myworkbooks_API.json
+    reason: 'delete API returns empty response and is already part of the previous api, cannot change it'
```
Comment (MyronFanQiu) [inline specification/applicationinsights/resource-manager/readme.md:105]: What's the meaning of previous apis? Consistent with `workbooks_API.json`?

**Comment by gardnerjr** [comment]
anyone know how to fix this specifically:

> Must not have duplicate name of x-ms-enum extension , make sure every x-ms-enum name unique. The duplicate x-ms-enum name is sharedtypekind
  at specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:412

the enum it is complaining about is exactly the same definition as the one in workbooks_api, and it doesn't complain about the categories enum that's in both of them?  (or maybe it just stops at that failure and doesn't show that one until i fix this one?)

i'd tried renaming it previously but that seemed to STILL say it was a duplicate across the 2 files?

File: specification/applicationinsights/resource-manager/readme.md (hunk 9 above)
Comment (gardnerjr) [reply specification/applicationinsights/resource-manager/readme.md:105]: yes'ish.  the endpoints/apis for workbooks and myworkbooks are
(1) exactly the same
(2) shipped at exactly the same time

i'm not sure why the S360 about myworkbooks only started showing up.  but now i have to add that, so i've been trying to fix it up.  it appears that years ago when workbooks was added, these rules either didn't exist or didn't apply, but now they do for myworkbooks.  so it keeps showing them as failures in lintdiff.  if there's a more correct way to do this, let me know.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json (hunk 8 above)
Comment (gardnerjr) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:448]: any/all, and it doesn't even need to be a real resource id.  it also technically supports things that aren't real resources, like "Azure monitor" and other areas that are cross cutting like azure backup, azure active directory, azure sentinel, etc.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json (hunk 7 above)
Comment (gardnerjr) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:444]: it is read only and assigned by the service when created.  it is _possibly_ the aad object id for the user?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json (hunk 6 above)
Comment (gardnerjr) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:435]: this inner tags inside properties is a legacy thing that existed before this resource type existed (when they were a proxy resource inside application insights, so it didn't support tags at the outer level).  in this case, it was just a list of tags, not a key:value pair like azure tags are.  (personally, we've talked about internally removing this internal tags entirely)

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json (hunk 5 above)
Comment (gardnerjr) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:412]: shouldn't be here.  removing.  the top level one is correct.  i think a lot of this never got fixed in the previous review of workbooks (which i wasn't directly involved in last time). 

i've already removed some of the other ones, like "workbookId" was here, but "sourceId" wasn't, and those still (incorrectly) appear in the workbooks_api.json.  I'd like to fix them but those would DEFINATELY show up as a breaking change, right?

**Comment by ravbhatnagar** [comment]
Signing off from ARM side.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json (hunk 7 above)
Comment (MyronFanQiu) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:429]: If it's readonly, could we add it?

**Comment by gardnerjr** [comment]
@MyronFanQiu  @ravbhatnagar is there anything else i need to do to complete this?  i'm not sure how/why it thinks the python sdk is breaking, it is showing things unrelated to what i touched (possibly this branch needs a merge or something?)

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json (hunk 7 above)
Comment (gardnerjr) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json:429]: kindof?  it is treated as readonly by the service, but if i tag it read only here am i going to have to touch all of the other examples/etc?  it was written in a way that it is returned in responses, and those responses can be updated and sent back in updates and the field will effectively ignored by the service, so you don't have to *remove* it to do a patch/etc.

>  code: 'READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST',
>                  params: [ 'userId', 'userId' ],
>                 message: 'ReadOnly property `"userId": "userId"`, cannot be sent in the request.',

i'm not sure there's a way to document system generated properties that are effectively "ignored" in a write?
i can mark it read only, but then have to delete it from the examples.

**Review by mmyyrroonn — APPROVED** [review]

**Pull request merged**